Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Generator.Equals.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            throw new Exception("Expected exactly one EquatableAttribute.");
+        }
```

Comment:
There's no need for any further validation. If the user applied multiple, they'll get a real compiler error, and you shouldn't crash the source generator. Just continue based on the first attribute.
```suggestion
        var equatableAttributeData = _context.Attributes[0];
```